packages/vitest/src/node/core.ts

```diff
@@ -163,7 +163,17 @@ export class Vitest {
     const projects = await this.resolveWorkspace(cliOptions)
     this.projects = projects
     this.resolvedProjects = projects
-    const filteredProjects = toArray(resolved.project)
+    const filteredProjects = toArray(resolved.project).reduce((acc, s) => {
+      if (s.includes('*')) {
+        const pattern = new RegExp(`^${s.replace(/\*/, '.*')}`)
```

Comment:
Coincidentally playwright just released `--project` flag with wildcard in the latest 1.42.0 https://github.com/microsoft/playwright/releases/tag/v1.42.0

Looking at what playwright did in
- https://github.com/microsoft/playwright/pull/29537
  (see `wildcardPatternToRegExp` in `packages/playwright/src/runner/projectUtils.ts`)

technically we should also escape special characters in `s` first. I think we can just copy over whole `wildcardPatternToRegExp`?

Reply:
Thanks for the advise, I'll take a look